extensions/daemon.md

```diff
@@ -0,0 +1,130 @@
+---
+description: Golem daemon extensions
+---
+
+# Overview
+
+Autostart extensions are executed during startup of the daemon process. The "autostart" option can be configured
+via `yagna`'s CLI or by placing an [extension manifest](#manifest) next to the extension binary.
+
+Please refer to the [designated directory list](README.md#designated-directories) to find out where to install an extension.
+
+## Autostart CLI
+
+Assuming that there is a `yagna-xchange` extension installed, we can execute that extension on daemon startup by typing the following:
+
+```bash
+yagna extension register xchange usd
+```
+
+The command above will create a manifest file next to the binary, informing the daemon that `yagna-xchange` should be executed on startup with appropriate arguments.
+
+`yagna extension list` should now report an output similar to:
+
+```bash
+┌────────────┬─────────────┬─────────────────────────────────────────────┬────────┐
+│    name    │  autostart  │  path                                       │  args  │
+├────────────┼─────────────┼─────────────────────────────────────────────┼────────┤
+│   xchange  │      x      │  /home/user/.local/bin/yagna-xchange        │  usd   │
+└────────────┴─────────────┴─────────────────────────────────────────────┴────────┘
+```
+
+In order to disable autostart, run
+
+```bash
+yagna extension unregister <name>
+```
+
+`yagna extension list` should now report an output similar to:
+
+```bash
+┌────────────┬─────────────┬─────────────────────────────────────────────┬────────┐
+│    name    │  autostart  │  path                                       │  args  │
+├────────────┼─────────────┼─────────────────────────────────────────────┼────────┤
+│   xchange  │             │  /home/user/.local/bin/yagna-xchange        │  usd   │
+└────────────┴─────────────┴─────────────────────────────────────────────┴────────┘
+```
+
+Since each call to `extension list` performs an extension binary lookup, we can tell that the process involves
```

Comment:
@mfranciszkiewicz I don't understand what this sentence tells the user... 

do we just want to convey a hint that each run of "extension list" triggers parsing of each manifest file? does this information add anything from the extension developer's pov?

Reply:
Maybe it should be rephrased that each call to `extension ...` actually tries to find all available extensions anew, including their configuration (manifests), and no caching is involved. 

It also states that when you modify a manifest file, `extension list` will reflect that change immediately. The daemon does not have to be running, which is helpful for deploying extensions